### PR TITLE
Update imports for Rancher Components

### LINKED ISSUE #5935: [@rancher/components] Replace local Dashboard components with Rancher Components

### LINKED ISSUE #7160: Allow use of aliases and scoped packages in imports within all the folders
# Description

Allow to use aliases and scoped packages in imports in e2e and other paths.

Occurrences:
- ~@rak-phillip `@rancher/components` in [E2E tests](https://github.com/rancher/dashboard/actions/runs/2922903499/jobs/4660940044)~ solved with rancher/dashboard#7425
- @n313893254 `@pkg` [when using unit tests](https://github.com/rancher/dashboard/actions/runs/3334501675/jobs/5517530061) 

## Context

Due to a proper configuration in any other folder out of `./shell`, it is not possible to use aliases or scoped packages. This forces the use of relative paths which is quite inconvenient.
E.g.:
- [In `pkg/harvester`](https://github.com/rancher/dashboard/blob/master/pkg/harvester/edit/kubevirt.io.virtualmachine/VirtualMachineVolume/type/existing.vue#L9) `import { HCI } from '../../../../types';`

### PR DESCRIPTION
**Pull request by rak-phillip**
### Summary

This updates all imports for Rancher Components to use the scoped `@rancher/components` npm package instead of the `@components` alias. We do this to encourage clean separation between Rancher Components, Shell, and other packages. This change also allows us to develop and release Rancher Components without affecting Rancher Dashboard development. 

closes #5935
closes #7160

### Occurred changes and/or fixed issues

* Update import statements to target `@rancher/components`
* Remove unused instances of `@components` alias

### Areas or cases that should be tested

Anything component or page that includes a

* Badge State
* Banner
* Card
* Checkbox
* Labeled Input
* Radio Group
* Text Area
* Toggle Switch
* Labeled Tooltip

Epinio, Harvester, and Rancher Shell should also be tested for regressions

### Areas which could experience regressions

I noticed some whitespace regressions in 

* cypress/e2e/po/components/checkbox-input.po.ts
* cypress/e2e/po/components/labeled-input.po.ts
* shell/components/__tests__/CruResource.test.ts

There were no observable visual changes in the associated components that these tests address. It looks like there's some sort of webpack config that was preserving whitespace in Rancher Dashboard and I'm not so sure if we want to retain this behavior.

### CODE REVIEW

**Comment by rak-phillip** [comment]
@mantis-toboggan-md I'm requesting your review so that we could validate that Harvester is still functioning properly

@richard-cox I'm requesting the same as above but for Epinio

@nwmac @cnotv I'm looking for feedback on the rest of the changes represented here 😁

Let me know if I've missed an important component/area so that I can get the proper person involved to verify and regress this change.

**Comment by rak-phillip** [comment]
> @rak-phillip does this solve the issue with the alias import for Jest and Cypress? #7160

Yes, we have a closing tag for this issue in the PR description 😁

> Are development procedure changed since we are using now a pointed version, e.g. we need to run and link the local build when developing? If yes, this would require documentation to be clear for people how to proceed.

This is a good point, I can work on updating development docs. 

> PS: There's some components which are going to be merged after the tech-debt, so if you are fine with that, it may be worthy to wait for that rebase. At least for the trivial cases where you are not mentioned sweat_smile

We will rebase as needed to incorporate changes 🙂

**Review by mantis-toboggan-md — DISMISSED** [review]
This all seems to work fine with harvester: I tested by importing a harvester cluster and loading the `harvester-latest` plugin.

**Comment by cnotv** [comment]
I tried to run the project with the library and it seems work fine.

We should consider excluding from hot reloading the folder or we can simply wait to detach it into another repository, as currently it reload the whole app on file changes without produce actual changes.

Another point to remind is the styling of some components are set global inside the `shell/assets/styles`, which we may want to move within the library itself or at least plan it, since they may affect more components. Currently in the configuration I can see only the variables and mixins.

I tried to use `yarn link` and `yarn link "@rancher/components"`, then written changes but I must have been failing something as I see nothing change 😬

**Comment by rak-phillip** [comment]
@scures can you validate and regress this PR against Epinio?

**Comment by rak-phillip** [comment]
> Haven't had a chance to look through this, can we pause until Monday? There needs to be something in place to address broken drone builds when components change, either by process or automation

I understand that this is a potentially destructive change. This is why I have requested a review from a large pool. We haven't merged after a single approval and I don't anticipate that we merge before this PR has received a thorough review from key members of the team. 

With that said, I would like to continue collecting feedback from others until you get time to review. I would like identify and address any potential regressions if any arise.

@richard-cox

**Comment by cnotv** [comment]
@rak-phillip I want to take a second look installing everything, as it must be an issue from my side. Yesterday I realized to have been installing packages in `./shell`, which forced the app to use old libraries (e.g. `rancher/icons`) and did not allow me to see changes. This is planned to be added in the troubleshooting section https://github.com/rancher/dashboard/issues/7464

**Review by richard-cox — CHANGES_REQUESTED** [review]
Disclaimer - I've done no testing of the actual changes, just a high level review.

General Comments
- I'm a little nervous making changes to `pkg/harvester`, given that this is now the legacy version that won't go through any testing for a while. It's unavoidable though, if the changes here don't happen it's totally broken.
- Might be worth testing the app/plugin creators. These might be broken atm, but would be good to validate we get the same error (there should be a github action that does this called `check-plugins`, but it's currently disabled)

Blocker
- This will currently (situational) break drone builds of the dashboard that occur when code merges to certain branches. If a PR contains changes to rancher components they won't get picked up by drone. There's a number of ways around this, needs discussing
  1) Make a separate PR that contains required changes to rancher components. This is merged first and a build created from it. The new rancher component package version can then be added to the second PR. This is quite a manual process and would make PR reviews tricky.
  2) Update build process to run a situational rancher components build and publish it before the main build occurs. This is neat... but the new rancher components version would need to get applied to the shell's package.json

File: shell/package.json
Comment (rak-phillip) [reply shell/package.json:52]: I comfortable with graduating from alpha. I had originally tagged it as such to serve as a test bed for Rancher Desktop integration. We haven't experienced any issues with the library since its integration.

**Comment by mantis-toboggan-md** [comment]
> * I'm a little nervous making changes to `pkg/harvester`, given that this is now the legacy version that won't go through any testing for a while. It's unavoidable though, if the changes here don't happen it's totally broken.

Are we actively using the pkg/harvester code in `master`? I thought we were pulling the already-built 1.0.3 plugin from the CDN...so the only time we'd be building a new version is when we're intentionally patching (and therefore actually testing) the legacy plugin

**Review by Sean-McQ — CHANGES_REQUESTED** [review]
I had to change the "@rancher/components" dependency in shell to "0.1.0-alpha.8" in order to get live-reload to work (with Phil's assistance) but once that's changed, it looks good to me.

**Comment by cnotv** [comment]
@rak-phillip I got [the same error of the build on E2E](https://github.com/rancher/dashboard/actions/runs/3567790741/jobs/5995907214) when I try to build the library.

**Comment by rak-phillip** [comment]
> @rak-phillip I got [the same error of the build on E2E](https://github.com/rancher/dashboard/actions/runs/3567790741/jobs/5995907214) when I try to build the library.

@cnotv yes, I should have called this out in a comment, this wasn't an issue before performing a rebase. 

You are seeing this error because `DraggableZone` requires an update. I linked issue #7567 and raised PR #7567 to address.

**Comment by rak-phillip** [comment]
Waiting for #7611 to merge. This will allow us to work with updated imports in storybook without the needs of workarounds.

File: shell/package.json
Comment (cnotv) [inline shell/package.json:52]: Current version is `0.1.1`. As this seems a nuisance for each bump, should we think about not pin the versions or what would you recommend?

**Comment by cnotv** [comment]
I have been finally able to update the components, prior manually set the library version and the dependency version to the same value. Personally further investigations to simplify the process may be done in a separated issue, but some notes about this written in bold are definitely required.

**Comment by rak-phillip** [comment]
> I have been finally able to update the components, prior manually set the library version and the dependency version to the same value. Personally further investigations to simplify the process may be done in a separated issue, but some notes about this written in bold are definitely required.

Please contribute to the Rancher Components docs with improvements based off your experience 🙂

https://github.com/rancher/dashboard/tree/master/pkg/rancher-components#developing-and-debugging

**Review by richard-cox — APPROVED** [review]
After discussing with team I'm removing my blocker review (https://github.com/rancher/dashboard/pull/7425#pullrequestreview-1187901142)
We should document the process on how to merge dashboard changes that also require changes to rancher components (and share with team). If we hit issues with this process we can work out how automate at that point.
 
**Disclaimer - I'll leave the granular code review and testing to the other developers (please don't merged until someone else has give the green tick of goodness)**

**Comment by rak-phillip** [comment]
closing as stale

**Pull request closed without merging**
